src/xdsl/dialects/llvm.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+from dataclasses import dataclass
+
+from xdsl.irdl import ParameterDef, AnyAttr, irdl_op_builder, irdl_attr_definition, AttributeDef, OperandDef, ResultDef, irdl_op_definition
+from xdsl.ir import MLContext, ParametrizedAttribute, TYPE_CHECKING, Attribute, Operation
+from typing import overload
+from xdsl.dialects.builtin import StringAttr, ArrayOfConstraint, ArrayAttr
+
+if TYPE_CHECKING:
+    from xdsl.parser import Parser
+    from xdsl.printer import Printer
+
+
+@dataclass
+class LLVM:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_attr(LLVMStructType)
+
+        self.ctx.register_op(LLVMExtractValue)
+        self.ctx.register_op(LLVMInsertValue)
+
+
+@irdl_attr_definition
+class LLVMStructType(ParametrizedAttribute):
+    name = "llvm.struct"
+
+    struct_name: ParameterDef[StringAttr]
+    types: ParameterDef[ArrayAttr[AnyAttr()]]
+
+    # bitmask = ParameterDef(StringAttr)
```

Comment:
Is this something that plan on being added? Should we remove the comment or add documentation why we keep this?

Reply:
This is the part of the type in MLIR, that defines the memory layout. AFAIK, it can be `packed` or `opaque`. For all three parameters, it is the case that they are optional, but we do not (yet?) support `OptionalParameterDef`. So I decided to add the other two with an empty string referring to a struct without a name, because this was the main functionality i needed. This is mainly here in order to remind us that the type is not fully mirrored yet. I added docu to reflect that fact.